src/server/repositories/UrlRepository.ts

```diff
@@ -228,14 +229,16 @@ export class UrlRepository implements UrlRepositoryInterface {
     state: string | undefined,
     isFile: boolean | undefined,
   ): Promise<UrlDirectoryPaginated> {
+    // Extract shortUrls with regex
+    const newQuery = extractShortUrl(query)
```

Comment:
I'm not convinced that it is the server's responsibility to correct user input. @yong-jie wdyt?

Reply:
agreed in general that server shouldnt correct user input. in this control flow they're supposed to be searching for keywords, and a shortUrl isnt a keyword. 

Extracting the shortURL here would mean that either 1) our "keyword" header is incorrect, or  2) this function is doing things that it's not supposed to do. Both scenarios are undesirable

That being said, it's strange that we didnt give them a way to explicitly search a shortlink. I think even we as developers are finding this weird, which is why we're anticipating user mistakes like these. 

It might be worth to discuss whether we should have an option to search shortlinks too(?)

Reply:
I don't know. we are taking on some complexity, and the gain is somewhat questionable. should the API be that forgiving?